spec/constraints/role_constraint_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe RoleConstraint do
+  describe '#matches?' do
+    let(:request) { double }
+    let(:user) { create(:user, :admin) }
+
+    subject { RoleConstraint.new('admin') }
+
+    before :each do
+      env = double
+      allow(request).to receive(:env).and_return(env)
+      allow(env).to receive(:[]).with('warden').and_return(env)
+      allow(env).to receive(:user).and_return(user)
+    end
+
+
+    context 'when role matches' do
+      it 'returns true' do
+        expect(subject.matches?(request)).to eq(true)
```

Comment:
minor thing, actually just a preference, I think it reads better as:
`expect(subject.matches?(request)).to be true`